backend/src/appointment/routes/api.py

```diff
@@ -456,6 +458,45 @@ def delete_my_appointment(id: int, db: Session = Depends(get_db), subscriber: Su
     return True
 
 
+@router.post('/apmt/{id}/cancel')
+def cancel_my_appointment(
```

Comment:
Like @devmount mentioned we'll need to call calendar.delete_event, we don't have a good one-line way to initialize that right now, so feel free to copy and paste from another function.